### PR TITLE
Add input consumer for beatmap

### PR DESCRIPTION
**Pull request by zijunm**

### CODE REVIEW

File: Holo-annie-jam/Managers/RhythmManager/BeatmapPlayer.cs [hunk 1]
```diff
@@ -104,22 +105,59 @@ public void Pause() {
...
             .Skip(this.firstVisibleEventIdx)
             .Take(this.nextVisibleEventIdx - this.firstVisibleEventIdx);
     }
 
+    /// <summary>
+    /// Takes player input, assuming input was at current time, and matches it to a rhythm event in the beatmap to
+    /// consume it. If no match was found, one of NoHit and Miss will be returned.
+    /// </summary>
+    public BeatmapHitResult ConsumePlayerInput(InputType inputType) {
+        long tick = MediaPlayer.PlayPosition.Ticks;
```
[the remark this answers is not in the document]
Comment (zijunm) [reply Holo-annie-jam/Managers/RhythmManager/BeatmapPlayer.cs:121]: this one is retrievable anywhere in the game using MediaPlayer.PlayPosition.Ticks, shouldn't be an issue unless we move off MediaPlayer

File: Holo-annie-jam/Managers/RhythmManager/BeatmapPlayer.cs [hunk 2]
```diff
@@ -104,22 +105,59 @@ public void Pause() {
...
         return this._beatmap.RhythmEvents
             .Skip(this.firstVisibleEventIdx)
             .Take(this.nextVisibleEventIdx - this.firstVisibleEventIdx);
     }
 
+    /// <summary>
+    /// Takes player input, assuming input was at current time, and matches it to a rhythm event in the beatmap to
+    /// consume it. If no match was found, one of NoHit and Miss will be returned.
+    /// </summary>
+    public BeatmapHitResult ConsumePlayerInput(InputType inputType) {
```
[the remark this answers is not in the document]
Comment (zijunm) [reply Holo-annie-jam/Managers/RhythmManager/BeatmapPlayer.cs:120]: I forgor 💀

File: Holo-annie-jam/Managers/RhythmManager/BeatmapPlayer.cs [hunk 3]
```diff
@@ -104,22 +105,59 @@ public void Pause() {
             }
         }
         while (this.nextVisibleEventIdx < this._beatmap.RhythmEvents.Length &&
-               this._beatmap.RhythmEvents[this.nextVisibleEventIdx].Tick < MediaPlayer.PlayPosition.Ticks + this.VisibleTimespanTicks) {
+                this._beatmap.RhythmEvents[this.nextVisibleEventIdx].Tick < MediaPlayer.PlayPosition.Ticks + this.VisibleTimespanTicks) {
```
[the remark this answers is not in the document]
Comment (zijunm) [reply Holo-annie-jam/Managers/RhythmManager/BeatmapPlayer.cs:108]: I just wanted to change it from lining up to 2 tabs, which is more consistent

**Review by pseudobear — APPROVED** [review]
works for now

**Pull request merged**